src/main/java/com/scaler/lld/design/creational/builder/database/Database.java

```diff
@@ -9,62 +9,65 @@ public class Database {
     private int port;
     private DatabaseType type;
 
-    // Step 1 - Hide the constructor
     private Database() {
     }
 
+    // Step 1 - Hide the constructor
+    private Database(String name, String host, int port, DatabaseType type) {
+        this.name = name;
+        this.host = host;
+        this.port = port;
+        this.type = type;
+    }
+
     // Add a static method to get the builder object
     public static DatabaseBuilder builder() {
-        return new DatabaseBuilder();
+        return new DatabaseBuilder(new Database());
     }
-    
+
     // Step 2 - Create a static inner class with same fields as the outer class
     public static class DatabaseBuilder {
-        private String name;
-        private String host;
-        private int port;
-        private DatabaseType type;
+        private Database database;
 
+        DatabaseBuilder(Database database) {
+            this.database = database;
+        }
```

Comment:
Let's remove the constructor with the database since it can be called from different classes.
You can create just a new instance in a no-argument constructor.

Reply:
got it, updated the code as per your suggestion, Please check